dask/bytes/core.py

```diff
@@ -141,7 +143,7 @@ def write_bytes(data, urlpath, name_function=None, compression=None,
         raise NotImplementedError("Unknown protocol for writing %s (%s)" %
                                   (protocol, urlpath))
 
-    files = open_files_write(paths, **storage_options)
+    files = [(open_files_write, p, storage_options) for p in paths]
     out = [delayed(write_block_to_file)(v, f, compression, encoding)
            for (v, f) in zip(data, files)]
```

Comment:
Building more complex tasks to put in delayed objects 

```python
keys = ['write-block-%s' % tokenize(path, ...) for path in paths]
return [Delayed(key, dasks=[{key: (write_block_to_file, v, 
                                    (apply, open_files_write, (p,), storage_options), 
                                    compression, encoding) ]) 
                for key, v, f in zip(keys, data, files)]
```

This marries file creation and writing into a single task but also avoids putting the file creation inside of `write_bytes_to_file`, which still just takes `bytes + file -> None`